dfir_lang/src/graph/ops/batch.rs

```diff
@@ -22,58 +19,5 @@ pub const BATCH: OperatorConstraints = OperatorConstraints {
     ports_inn: None,
     ports_out: None,
     input_delaytype_fn: |_| None,
-    write_fn: |wc @ &WriteContextArgs {
-                   root,
-                   context,
-                   hydroflow,
-                   op_span,
-                   ident,
-                   is_pull,
-                   inputs,
-                   outputs,
-                   singleton_output_ident,
-                   ..
-               },
-               _diagnostics| {
-        let write_prologue = quote_spanned! {op_span=>
-            #[allow(clippy::redundant_closure_call)]
-            let #singleton_output_ident = #hydroflow.add_state(
-                ::std::cell::RefCell::new(::std::vec::Vec::new())
-            );
-
-            // TODO(mingwei): Is this needed?
-            // Reset the value to the initializer fn if it is a new tick.
-            #hydroflow.set_state_tick_hook(#singleton_output_ident, move |rcell| { rcell.take(); });
-        };
-
-        let vec_ident = wc.make_ident("vec");
-
-        let write_iterator = if is_pull {
-            // Pull.
-            let input = &inputs[0];
-            quote_spanned! {op_span=>
-                let mut #vec_ident = #context.state_ref(#singleton_output_ident).borrow_mut();
-                *#vec_ident = #input.collect::<::std::vec::Vec<_>>();
-                let #ident = ::std::iter::once(::std::clone::Clone::clone(&*#vec_ident));
-            }
-        } else if let Some(_output) = outputs.first() {
-            // Push with output.
-            // TODO(mingwei): Not supported - cannot tell EOS for pusherators.
-            panic!("Should not happen - batch must be at ingress to a loop, therefore ingress to a subgraph, so would be pull-based.");
-        } else {
-            // Push with no output.
-            quote_spanned! {op_span=>
-                let mut #vec_ident = #context.state_ref(#singleton_output_ident).borrow_mut();
-                let #ident = #root::pusherator::for_each::ForEach::new(|item| {
-                    ::std::vec::Vec::push(#vec_ident, item);
-                });
-            }
-        };
-
-        Ok(OperatorWriteOutput {
-            write_prologue,
-            write_iterator,
-            ..Default::default()
-        })
-    },
+    write_fn: IDENTITY_WRITE_FN,
```

Comment:
Maybe a brief comment like "all we need here is identity because scheduler will inherently produce batches"?

What was all the old stuff for?

Reply:
The previous `batch()` actually acted more like `prefix()`, in that it would replay the data on subsequent iterations. Which is a bug, except it couldn't show up until `repeat_n()` now existing

Reply:
Also previously it released `Vec<T>` instead of `T` which Shadaj clarified is incorrect